.github/workflows/test-analyze.yml

```diff
@@ -0,0 +1,98 @@
+# ==================================================
+# This is the main worflow for testing and analyzing
+# the code.
+# ==================================================
+
+name: CI
+
+on:
+  pull_request:
+    branches: [project/**, release/**]
```

Comment:
Let's also add main for now until we have figured out how we will use release branches.

Reply:
Sure

Reply:
Btw what will the project/** branches be for ??